Formula/scala.rb

```diff
@@ -27,22 +27,22 @@ class Scala < Formula
   end
 
   devel do
-    url "http://www.scala-lang.org/files/archive/scala-2.12.0-M4.tgz"
-    sha256 "2ea78303f1d1ba53ed4a1b905e22ea7a14556da0b31eb081f21a1b723bf672b7"
-    version "2.12.0-M4"
+    url "http://www.scala-lang.org/files/archive/scala-2.12.0-M5.tgz"
+    sha256 "fd260bd1d2fb7ba1e8003d40a463959a4111a3e4a25194e8d9e54e23cb39dd08"
+    version "2.12.0-M5"
 
     depends_on :java => "1.8+"
 
     resource "docs" do
-      url "http://www.scala-lang.org/files/archive/scala-docs-2.12.0-M4.zip"
-      sha256 "ccfbc72c5bfd3743b08f4ecfb3dea46cb2b982eaaba5ada24378a93c351f8c77"
-      version "2.12.0-M4"
+      url "http://www.scala-lang.org/files/archive/scala-docs-2.12.0-M5.zip"
+      sha256 "025d4db9755ffe0afc7254a105f95ca0ea6a77385d5c9b51a441aa8f34aefa5e"
+      version "2.12.0-M5"
```

Comment:
The version changes to `scala--docs-4.zip`, etc, which means we have resources in the cache that don't align with the expected devel version. That has the potential to cause issues when trying to cleanup the cache, etc.